compiler/rustc_const_eval/src/transform/check_consts/check.rs

```diff
@@ -846,6 +848,22 @@ impl<'tcx> Visitor<'tcx> for Checker<'_, 'tcx> {
                             {
                                 nonconst_call_permission = true;
                             }
+                            let call_kind = call_kind(
+                                tcx,
+                                self.param_env,
+                                callee,
+                                fn_args,
+                                *fn_span,
+                                call_source.from_hir_call(),
+                                None,
+                            );
+                            if let call_kind::CallKind::FnCall { fn_trait_id: _, self_ty } =
+                                call_kind
+                                && let FnDef(def_id, ..) = self_ty.kind()
+                                && tcx.is_const_fn_raw(*def_id)
```

Comment:
`is_const_fn_raw` only checks whether the function is `const`, and not whether it is `const`-stable. We'd need a test to check whether this would silently allow calling functions that are not const-stable.

Reply:
how do we do that kind of check ?

Reply:
I'm not sure if other parts of the const checking code already check for const stability, so the first step would be creating an auxiliary ui test where upstream has a const-unstable function and downstream tries to call that function without the feature enabled (following the snippet for this ICE). If it already passes, then great. If it doesn't, we can change this from `is_const_fn_raw` to `is_const_fn`